src/renderers/webgl/WebGLRenderStates.js

```diff
@@ -54,26 +61,26 @@ function WebGLRenderStates() {
 
 	let renderStates = new WeakMap();
 
-	function get( scene, camera ) {
+	function get( scene, renderCallNesting ) {
 
 		let renderState;
 
 		if ( renderStates.has( scene ) === false ) {
 
 			renderState = new WebGLRenderState();
-			renderStates.set( scene, new WeakMap() );
-			renderStates.get( scene ).set( camera, renderState );
+			renderStates.set( scene, [] );
+			renderStates.get( scene ).push( renderState );
 
 		} else {
 
-			if ( renderStates.get( scene ).has( camera ) === false ) {
+			if ( renderCallNesting >= renderStates.get( scene ).length ) {
```

Comment:
This code section is a bit hard to read. I wonder if it's not possible to write it simpler. Couldn't you just use a `Map`?

Reply:
I think it would be more confusing if a `Map` was used when the keys are indexes starting from zero.